docs/05-_Navigation_Interception.md

```diff
@@ -22,7 +22,39 @@ Example:
 
 As the value of the `target` attribute, you might consider one of the following keywords that have a special meaning in HTML:
 
-- `_top` - load the URL into the current browser tab or window
-- `_blank` - load the URL into a new browser tab or window
+-   `_top` - load the URL into the current browser tab or window
+-   `_blank` - load the URL into a new browser tab or window
 
 You can find more information on the `target` attribute at [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a).
+
+### Intercepting within-app navigation
+
+Before and after morphing the page upon navigation within your app, Palindrom dispatches bubbling navigation events. `palindrom-before-navigation` and `palindrom-after-navigation`. You can use these events to watch or even cancel navigation.
+
+To cancel Palindrom navigation, you can cancel `palindrom-before-navigation` event by calling `event.preventDefault()` inside your event handler. Then after you show your confirmation box or whatever prupose you canceled the navigation for, you can continue the navigation by calling `palindrom.morphUrl` to the same URL. To record the navigation URL, you can get it from `event.detail.href` of `palindrom-before-navigation`.
+
+#### Note
+
+When you proceed with the navigation, since your new navigation is still a navigation, you'll trigger your event handlers again. To handle this you may to create a local flag and set it true after you handled the expected navigation.
```

Comment:
The first sentence is hard to understand :)

WDYT about changing this paragraph to:

> A programmatic call to `palindrom.morphUrl` also triggers the navigation events listed above. To distinguish between the events caused by a user interaction and a call to  `palindrom.morphUrl`, you may use a local variable.